.github/workflows/delegator_generic_slsa3.yml

```
@@ -248,11 +248,13 @@ jobs:
   # cleanup deletes internal artifacts used by the delegator workflow
   cleanup:
     runs-on: ubuntu-latest
-    needs: [generate-provenance]
+    needs: [rng, generate-provenance]
+    env:
+      RNG: ${{ needs.rng.outputs.value }}
     steps:
       - uses: geekyeggo/delete-artifact@54ab544f12cdb7b71613a16a2b5a37a9ade990af # v2.0.0
         with:
           name: "*-${{ env.SLSA_PREDICATE_FILE }}"
```

Do you still want to delete all predicate files?
```suggestion
          name: "${{ env.RNG }}-${{ env.SLSA_PREDICATE_FILE }}"
```

---

oops good catch - i missed that one.